packages/server/test/integration/http_requests_spec.coffee

```diff
@@ -2610,6 +2610,90 @@ describe "Routes", ->
 
           expect(res.body).to.include("Cypress errored attempting to make an http request to this url")
 
+    context "blacklisted hosts", ->
+      beforeEach ->
+        @setup({
+          config: {
+            blacklistHosts: [
+              "*.google.com"
+              "shop.apple.com"
+              "cypress.io"
+              "localhost:6666"
+              "*adwords.com"
+            ]
+          }
+        })
+
+      it "returns 503 and custom headers for all hosts", ->
+        expectedHeader = (res, val) ->
+          expect(res.headers["x-cypress-matched-blacklisted-host"]).to.eq(val)
+
+        Promise.all([
```

Comment:
you can also use https://github.com/bahmutov/snap-shot-it#data-driven-testing to make the test simpler here